client/core/core.go

```diff
@@ -2621,10 +2642,19 @@ func (c *Core) register(dc *dexConnection, encKey, encKeyLegacy, acctPubB []byte
 			}
 
 			dcrID, _ := dex.BipSymbolID(regFeeAssetSymbol)
-			cid, _ := asset.DecodeCoinID(dcrID, dc.acct.feeCoin)
+			cid, err := asset.DecodeCoinID(dcrID, dc.acct.feeCoin)
+			if err != nil {
+				// feeCoin decoded from the server's error message above is invalid,
+				// use a dummy dcr fee coin.
+				var zeroHash chainhash.Hash
+				dc.acct.feeCoin = make([]byte, chainhash.HashSize+4)
```

Comment:
Quite unfortunate to have to do this.  Alternatives are fail registration entirely even though server recognizes us (not good), or do `dc.acct.feeCoin = []byte{1}` so the len > 0 checks that are all over the place pass.  The latter only would work if we ensure there's no other call to `asset.DecodeCoinID(dcrID, dc.acct.feeCoin)` where an error would cause a problem.

Thoughts @buck54321?  We have to check this error as it's server provided data.

Reply:
My read is that we can set it to a constant, even if invalid, as long as we don't error at `c.db.AccountPaid`. The `(*BoltDB).Account` methods use `acctInfo.Paid = len(acct.Get(feeProofKey)) > 0`, and we really don't use the address again, e.g. we'll never call `reFee` with this coin.

```Go
dc.acct.feeCoin, err = hex.DecodeString(msgErr.Message)
if err != nil {
	c.log.Errorf("Failed to decode fee coin from pre-paid account info message = %q, err = %w", msgErr.Message, err)
	dc.acct.feeCoin = []byte("DUMMY COIN")
} else {
	dcrID, _ := dex.BipSymbolID(regFeeAssetSymbol)
	cid, err := asset.DecodeCoinID(dcrID, dc.acct.feeCoin)
	if err != nil {
		c.log.Errorf("Failed to decode Decred coin ID for pre-paid account from feeCoin = %x, err = %w", dc.acct.feeCoin, err)
	} else {
		c.log.Infof("Recovered paid account for %s. Fee previously paid with %s", dc.acct.host, cid)
	}
}
```

as long as we can be certain that `(*BoltDB).AccountPaid` passes?

Reply:
> we'll never call `reFee` with this coin

Yeah, true.  That's where I saw that the content mattered but it's not used after this.  Let's go with the code you proposed @buck54321. **Except** we still need to catch `msgErr.Message == ""` 

https://play.golang.org/p/uB6PV_9wpc8

Reply:
Is there a reason to think `(*BoltDB).AccountPaid` might fail? Just looked through the implementation and doesn't seem to concern itself with the fee coin. Looks safe to me.